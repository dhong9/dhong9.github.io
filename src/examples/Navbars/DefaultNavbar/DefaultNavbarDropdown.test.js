import renderer from "react-test-renderer";
import DefaultNavbarDropdown from "examples/Navbars/DefaultNavbar/DefaultNavbarDropdown";

// Icons
import GitHubIcon from "@mui/icons-material/GitHub";
import FacebookIcon from "@mui/icons-material/Facebook";

// Mocks
jest.mock("@mui/material/Container", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Container</div>),
  };
});
jest.mock("@mui/material/Icon", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Icon</div>),
  };
});
jest.mock("@mui/material/Popper", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Popper</div>),
  };
});
jest.mock("@mui/material/Grow", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Grow</div>),
  };
});
jest.mock("@mui/material/Grid", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Grid</div>),
  };
});
jest.mock("@mui/material/Divider", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Divider</div>),
  };
});
jest.mock("@mui/material/Link", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <div>MUI Link</div>),
  };
});
jest.mock("components/MKBox", () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation((props) => {
      const { children } = props;
      return <div {...props}>{children}</div>;
    }),
  };
});
jest.mock("components/MKTypography", () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation((props) => {
      const { children } = props;
      return <div {...props}>{children}</div>;
    }),
  };
});
jest.mock("components/MKButton", () => {
  const { forwardRef } = jest.requireActual("react");
  return {
    __esModule: true,
    default: forwardRef(() => <button>Mock Button</button>),
  };
});

describe("DefaultNavbarDropdown", () => {
  it("renders with route", () => {
    const component = renderer.create(
      <DefaultNavbarDropdown
        routes={[{ label: "Navbar", route: "/default", icon: <GitHubIcon />, name: "Github" }]}
        collapse={false}
        icon={<FacebookIcon />}
        name="Routed Navbar"
      >
        <div>I have routes</div>
      </DefaultNavbarDropdown>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without route", () => {
    const component = renderer.create(
      <DefaultNavbarDropdown
        routes={[{ label: "Regular", icon: <GitHubIcon />, name: "Github" }]}
        collapse={false}
        icon={<FacebookIcon />}
        name="Not Routed Navbar"
      >
        <div>I have no routes</div>
      </DefaultNavbarDropdown>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
