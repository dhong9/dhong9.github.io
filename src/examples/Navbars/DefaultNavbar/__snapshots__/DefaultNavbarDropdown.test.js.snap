// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultNavbarDropdown renders with route 1`] = `
<div
  alignItems="baseline"
  color="dark"
  display="flex"
  mx={1}
  opacity={0.6}
  p={1}
  routes={
    [
      {
        "icon": <Memo />,
        "label": "Navbar",
        "name": "Github",
        "route": "/default",
      },
    ]
  }
  sx={
    {
      "cursor": "pointer",
      "userSelect": "none",
    }
  }
>
  <div
    color="inherit"
    lineHeight={1}
    sx={
      {
        "& *": {
          "verticalAlign": "middle",
        },
        "alignSelf": "center",
      }
    }
    variant="body2"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="FacebookIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
      />
    </svg>
  </div>
  <div
    color="dark"
    fontWeight="regular"
    sx={
      {
        "fontWeight": "100%",
        "ml": 1,
        "mr": 0.25,
      }
    }
    textTransform="capitalize"
    variant="button"
  >
    Routed Navbar
  </div>
  <div
    color="dark"
    ml="auto"
    variant="body2"
  >
    <div>
      MUI Icon
    </div>
  </div>
</div>
`;

exports[`DefaultNavbarDropdown renders without route 1`] = `
<div
  alignItems="baseline"
  color="dark"
  display="flex"
  mx={1}
  opacity={0.6}
  p={1}
  routes={
    [
      {
        "icon": <Memo />,
        "label": "Regular",
        "name": "Github",
      },
    ]
  }
  sx={
    {
      "cursor": "pointer",
      "userSelect": "none",
    }
  }
>
  <div
    color="inherit"
    lineHeight={1}
    sx={
      {
        "& *": {
          "verticalAlign": "middle",
        },
        "alignSelf": "center",
      }
    }
    variant="body2"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="FacebookIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
      />
    </svg>
  </div>
  <div
    color="dark"
    fontWeight="regular"
    sx={
      {
        "fontWeight": "100%",
        "ml": 1,
        "mr": 0.25,
      }
    }
    textTransform="capitalize"
    variant="button"
  >
    Not Routed Navbar
  </div>
  <div
    color="dark"
    ml="auto"
    variant="body2"
  >
    <div>
      MUI Icon
    </div>
  </div>
</div>
`;
